core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -595,6 +597,9 @@ oppia.controller('StateResponses', [
             $scope.trainingDataFeedback = '';
             $scope.trainingDataOutcomeDest = '';
 
+            // Inject textInputRulesService dynamically
+            textInputRulesService = $injector.get('textInputRulesService');
```

Comment:
Can we have a mapping from interaction ID to rules service in the feconf? The other alternative is to hard-code it in StateResponses.

Also, is retrieving interaction ID via "stateInteractionIdService.displayed" correct way to do it?